# Micrapis_genome_project

## Analysis scripts used in:
"De novo genome assemblies of the dwarf honey bee subgenus Micrapis: Apis andreniformis and Apis florea" (2025) Atma Ivancevic*, Madison Sankovitz*, Holly Allen, Olivia Joyner, Edward B. Chuong+, Samuel Ramsey+

\*co-first authors  
\+co-last authors

---

## Programs used:

- **Dorado** v0.7.2: https://github.com/nanoporetech/dorado  
- **Flye** v2.9.5: https://github.com/fenderglass/Flye  
- **Medaka** v2.0.0: https://github.com/nanoporetech/medaka  
- **Pilon** v1.24: https://github.com/broadinstitute/pilon  
- **BBMap/BBDuk** v38.05: https://sourceforge.net/projects/bbmap  
- **BWA-MEM** v0.7.15: https://github.com/lh3/bwa  
- **Samtools** v1.16.1: http://www.htslib.org/  
- **QUAST** v5.2.0: https://github.com/ablab/quast  
- **BUSCO** v5.7.1: https://busco.ezlab.org/  
- **Meryl** v1.4.1 & **Merqury** v1.3: https://github.com/marbl/merqury  
- **RepeatMasker** v4.1.7: https://www.repeatmasker.org/  
- **Tandem Repeats Finder** v4.0.9: https://tandem.bu.edu/trf/trf.html  
- **VSEARCH** v2.14.1: https://github.com/torognes/vsearch  
- **Pychopper** v2.5.0: https://github.com/nanoporetech/pychopper  
- **Minimap2** v2.22: https://github.com/lh3/minimap2  
- **HISAT2** v2.1.0: https://daehwankimlab.github.io/hisat2/  
- **BRAKER3**: https://github.com/Gaius-Augustus/BRAKER  
- **StringTie2**: https://github.com/gpertea/stringtie  
- **GffRead**: https://github.com/gpertea/gffread  
- **DIAMOND**: https://github.com/bbuchfink/diamond  

---

## Genome Assembly and Polishing Workflow

1. **ONT basecalling + demultiplexing**  
   `dorado_basecall_and_demux.sh`

2. **Assembly**  
   `flye_assembly.sh`  
   `medaka_polish.sh`  
   `pilon_polish.sh`

3. **Short-read preprocessing and mapping**  
   `bbduk_trim_genome.sh`  
   `bwa_mem_align.sh`  
   `samtools_sort_index.sh`

4. **Contig filtering and contamination screening**  
   `vsearch_filter_contigs.sh`  
   `fcs_adaptor.sh`  
   `fcs_gx.sh`

5. **Assembly QC**  
   `quast_report.sh`  
   `busco_genome.sh`  
   `meryl_merqury.sh`

---

## Transcriptome Assembly and Annotation Workflow

1. **ONT cDNA processing**  
   `dorado_basecall_cDNA.sh`  
   `pychopper_trim.sh`  
   `nanostat_cDNA_summary.sh`

2. **Illumina RNA-seq processing**  
   `bbduk_trim_rnaseq.sh`  
   `hisat2_index_align.sh`  
   `samtools_merge.sh`

3. **Transcript assembly and gene prediction**  
   `stringtie_assemble.sh`  
   `braker3_annotation.sh`  
   `diamond_align_orthodb.sh`  
   `gffread_filter.sh`  
   `busco_proteome.sh`

---

## Notes

- All software was run with default settings unless otherwise shown.  
- Protein homology for BRAKER3 was provided using OrthoDB Arthropoda dataset (v12):  
  [https://bioinf.uni-greifswald.de/bioinf/partitioned_odb12/Arthropoda.fa.gz](https://bioinf.uni-greifswald.de/bioinf/partitioned_odb12/Arthropoda.fa.gz)

---

## Raw and processed data are publicly available in the following repositories:

#### NCBI BioProject:
_Apis andreniformis_
1. Genome sequencing and assembly: NCBI BioProject [PRJNA1217036](https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA1217036)
2. Transcriptome sequencing: NCBI BioProject [PRJNA1217108](https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA1217108)

_Apis florea_
1. Genome sequencing and assembly: NCBI BioProject [PRJNA1217041](https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA1217041)
2. Transcriptome sequencing: NCBI BioProject [PRJNA1217110](https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA1217110)

#### GenBank:
_Apis andreniformis_ assembly ASM4859352v1: [GCA_048593525.1](https://www.ncbi.nlm.nih.gov/datasets/genome/GCA_048593525.1/)

_Apis florea_ assembly ASM4859348v1: [GCA_048593485.1](https://www.ncbi.nlm.nih.gov/datasets/genome/GCA_048593485.1/)

#### Zenodo:
Gene annotations generated by BRAKER3 are available on [Zenodo](https://doi.org/10.5281/zenodo.15048194).
